File: backend/LMProject/src/main/java/fpt/aptech/LMProject/entities/Book.java
package fpt.aptech.LMProject.entities;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.Table;

import com.fasterxml.jackson.annotation.JsonIgnore;

import lombok.Data;

@Entity
@Table(name = "Book")
@Data
public class Book{

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "BookID")
    private int bookID;

    @Column(name = "Title")
    private String  title;

    @Column(name = "Author")
    private String  author;

    @Column(name = "Description")
    private String description;

    @Column(name = "Price")
    private String price;

    // @Column(name = "CategoryID")
    // private int categoryID;
    
    @Column(name = "StockQuantity")
    private String stockQuantity;

    @Column(name = "ImageURL")
    private String imageURL;

    @JsonIgnore
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "CategoryID")
    private Category category;

}

File: backend/src/main/kotlin/com/example/kotlindemo/model/Book.kt
package com.example.kotlindemo.model

import java.math.BigDecimal
import javax.persistence.*
import javax.validation.constraints.NotBlank

@Entity
@Table(name = "book")
class Book {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var bookid: Long? = null

    var title: String? = null
    var author: String? = null
    var categoryid: Int? = null
    var price: BigDecimal? = null
    var description: String? = null
    var stockquantity: Int? = null
    var isbn: String? = null
    var publishedDate: String? = null
    var image: String? = null
    var languageID: Int? = null
    var publisherID: Int? = null
    var weight: Int? = null
    var page: Int? = null
    var dimensions: String? = null
    var featured: Boolean? = null
    var bestseller: Boolean? = null // getters and setters
}
